geode-core/src/main/java/org/apache/geode/management/internal/configuration/domain/DeclarableTypeInstantiator.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.management.internal.configuration.domain;
+
+import java.util.Properties;
+
+import org.apache.geode.cache.Cache;
+import org.apache.geode.cache.Declarable;
+import org.apache.geode.cache.configuration.ConfigTypeInstantiator;
+import org.apache.geode.cache.configuration.DeclarableType;
+import org.apache.geode.cache.configuration.ParameterType;
+import org.apache.geode.internal.ClassPathLoader;
+
+public class DeclarableTypeInstantiator implements ConfigTypeInstantiator<DeclarableType> {
+
+  private final Cache cache;
+
+  public DeclarableTypeInstantiator(Cache cache) {
+    this.cache = cache;
+  }
+
+  @Override
+  public <T> T newInstance(DeclarableType declarableType) {
```

Comment:
Maybe `T extends Declarable`

Reply:
Doesn't seem to be syntactically possible, but my Generic foo is weak...

Reply:
you'll have to define the `ConfigTypeInstantiator` as follows:
`public interface ConfigTypeInstantiator<T,Q>{
<V extends Q> V newInstance(T type)
}`

and then the `DeclarableTypeInstantiator` can become:
`public class DeclarableTypeInstantiator implements ConfigTypeInstantiator<DeclarableType,Declarable>{
@Override
public <T extends Declarable> T newInstance(DeclarableType type){...
...
...
}`

Reply:
Unfortunately that also doesn't get to what you'd like. For this to work, the concrete method declaration would then need to be `public <V> V newInstance(ConfigTypeInstantiator<DeclarableType, Declarable> instantiator) {`. However that then requires the `Declarable` class to be visible to this module. The only way I see to get this to compile, is to replace `Declarable` with `?` which ends up again not restricting the returned type.